### PR TITLE
Cmake: cleanup the use of udev and blkid in target_link_lib()

### PR DESCRIPTION
**Pull request by wjwithagen**
Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

File: src/librbd/CMakeLists.txt [hunk 1]
```diff
@@ -107,9 +107,12 @@ target_link_libraries(librbd LINK_PRIVATE
   cls_journal_client 
   common
   pthread
-  udev
   ${CMAKE_DL_LIBS}
   ${EXTRALIBS})
+if(HAVE_UDEV)
+  target_link_libraries(librbd LINK_PRIVATE
+    ${UDEV_LIBRARIES})
+endif()
```
Comment (tchaikov) [inline src/librbd/CMakeLists.txt]: @wjwithagen i think we can drop udev here.

Comment (wjwithagen) [inline src/librbd/CMakeLists.txt]: @tchaikov 
Let try...

**Comment by tchaikov** [comment]
pushed to ceph-ci, and shaman builds fine.

**Comment by wjwithagen** [comment]
@tchaikov 
Thanx,

**Pull request merged**